### PR TITLE
feature: disable root path check when serve is false

### PR DESCRIPTION
**Pull request by nimesh0505**
### Summary
This MR addresses the issue [#460](https://github.com/fastify/fastify-static/issues/460) where the `checkRootPathForErrors` function still requires a valid path to root even when `serve: false` is set. 

### Changes
- Disabled `checkRootPathForErrors` when `serve: false`.
- Added a unit test to confirm that the root path check is bypassed when `serve: false`.

### Tests
- Added a test in `test/static.js` to verify that `checkRootPathForErrors` is not called when `serve: false`.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]
I don't think simply disable the check is the solution.

1. It may fails to serve file because there is no `root` provided. You can see the `root` path is passing everywhere inside the codebase. You must provide test to ensure it will works.
2. It can leads to security issue because it means by default the serving of files can be escaped to everywhere. You should at least default the `root` to CWD and update the read me.

**Comment by Uzlopak** [comment]
My remark from three days ago was still not addressed.

> Imho this PR does not take in account, that if no general root path is defined, that `rootPath` parameter of `sendFile` becomes mandatory.
> 
> https://github.com/fastify/fastify-static/blob/39e89ea4852a9de5258d14930b4e2ac06d4b91d1/index.js#L84

**Comment by climba03003** [comment]
Note that please do not open another PR for updates or address comment.
It is hard to follows.

You should push commit to your branch to address the comments.

**Comment by nimesh0505** [comment]
@climba03003 Can you review this PR?

File: README.md [hunk 1]
```diff
@@ -435,10 +435,36 @@ Assume this structure with the compressed asset as a sibling of the un-compresse
...
 ```
 
-#### Disable serving
+#### Disable serving and CWD behavior
 
 If you would just like to use the reply decorator and not serve whole directories automatically, you can simply pass the option `{ serve: false }`. This will prevent the plugin from serving everything under `root`.
 
+When `serve: false` is passed:
+
+1. The plugin will not perform the usual directory existence check for the `root` option.
```
Comment (Uzlopak) [inline README.md:444]: too much information about internals

Comment (Uzlopak) [reply README.md:444]: devs dont want to know what the plugin does under the hood. here people want to know that sendFile will then send files from the current working directory.

File: index.js [hunk 2]
```diff
@@ -408,7 +413,7 @@ function normalizeRoot (root) {
   return root
 }
 
-function checkRootPathForErrors (fastify, rootPath) {
+function checkRootPathForErrors (fastify, rootPath, skipExistenceCheck) {
```
Comment (climba03003) [inline index.js:416]: I am think is it necessary to skip the check of `root` path?
You should never use a invalid root path, defaulting to `cwd` is a trade-off for security reason.

All of your served file should be contained within to the `root` and prevent path escape.

Comment (nimesh0505) [reply index.js:416]: @climba03003 Thank you for your feedback on the security implications. Before implementing changes, I would like to confirm the approach that we remove the `skipExistenceCheck` parameter from the `checkPath` and `checkRootPathForErrors` functions since we need perform full validation of the path, including existence checks.

Comment (climba03003) [reply index.js:416]: You can skip for cwd, but you must provide a valid root.
No matter other setting.

**Comment by mcollina** [comment]
@nimesh0505 @simoneb ping

**Comment by simoneb** [comment]
Unfortunately we cannot progress on this PR, so I'm closing it.
